src/cljs/rems/administration/create_catalogue_item.cljs

```diff
@@ -94,15 +96,17 @@
     (merge {:state "disabled"} request)
     request))
 
-(defn- create-catalogue-item [request]
-  (post! "/api/catalogue-items/create" {:params (create-request-with-state request)
-                                        ;; TODO error handling
-                                        :handler (fn [resp] (dispatch! "#/administration/catalogue-items"))}))
+(defn- create-catalogue-item! [{:keys [request on-pending on-success on-error]}]
+  (when on-pending (on-pending))
+  (post! "/api/catalogue-items/create" (merge
+                                        {:params (create-request-with-state request)
+                                         :handler on-success}
+                                        (when on-error {:error-handler on-error}))))
```

Comment:
`when on-error` is unnecessary and can be removed from all API calls. The `post!` will handle it automatically if it is nil. (Actually it would be even better if was an error if the error handler is accidentally missing.)

Reply:
Fixed